### PR TITLE
Start move to DBConnection 2.0

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: integration_test/sql/transaction.exs [hunk 1]
```diff
@@ -224,20 +225,6 @@ defmodule Ecto.Integration.TransactionTest do
     catch_error(PoolRepo.query!("savepoint foobar"))
   end
 
-  test "log raises after begin, drops the whole transaction" do
```
[the remark this answers is not in the document]
Comment (fishcakez) [reply integration_test/sql/transaction.exs]: correct, we considered this is a bug, and not a feature.

File: lib/ecto/adapters/postgres/connection.ex [hunk 2]
```diff
@@ -93,14 +91,12 @@ if Code.ensure_loaded?(Postgrex) do
       case DBConnection.execute(conn, query, params, opts) do
         {:ok, _} = ok ->
           ok
-        {:error, %ArgumentError{} = err} ->
-          {:reset, err}
```
[the remark this answers is not in the document]
Comment (fishcakez) [reply lib/ecto/adapters/postgres/connection.ex]: :+1:

File: lib/ecto/adapters/postgres/connection.ex [hunk 3]
```diff
@@ -93,14 +91,12 @@ if Code.ensure_loaded?(Postgrex) do
...
+        {:ok, _, _} = ok ->
+          ok
         {:error, %Postgrex.Error{postgres: %{code: :feature_not_supported}} = err} ->
           {:reset, err}
```
[the remark this answers is not in the document]
Comment (fishcakez) [reply lib/ecto/adapters/postgres/connection.ex]: Yes. This is the case where a query is already prepared on a connection, we execute it, but a migration/column type change occurred. In this situation we must close and prepare the query on that connection again if we want to repeat it. As this signals a type change its very likely we also want to reset the global cache.

**Comment by josevalim** [comment]
Merged!

**Comment by anthonator** [comment]
👏 👏 👏

**Pull request closed without merging**